### PR TITLE
x86jit: Force INF * 0 to +NAN

### LINKED ISSUE #12519: Ghost In The Shell - Stand Alone Complex (ULUS10020) - Black Textures and missing screens.
### What happens?
Some textures are black and some screens are completely missing (also black screen). The missing screens includes the green text intro at the beginning and chapters names. It's happening since 1.6.3-454. It seems #11380 broke it.

ULUS10020_00000
ULUS10020_00001

### What *should* happen?
The proper image should be as follows:

ULUS10020_00000
ULUS10020_00001

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.
It happens on any version after 1.6.3-454.
Windows 7 64-bit, GTX 750 ti, 8gb RAM.

### PR DESCRIPTION
**Pull request by unknownbrackets**
Fixes #12519 - this is needed for some graphics to render properly.  Seems to already happen on ARM, so no change to armjit.  Thanks to @iota97 for the investigation.

@iota97 does this help #12519?

Could also help #12436.

-[Unknown]

### CODE REVIEW

**Comment by iota97** [comment]
Yeah, #12519 is fixed :)

File: Core/MIPS/MIPSInt.cpp [hunk 1]
```diff
@@ -974,7 +974,14 @@ namespace MIPSInt
 		{
 		case 0: F(fd) = F(fs) + F(ft); break; // add.s
 		case 1: F(fd) = F(fs) - F(ft); break; // sub.s
-		case 2: F(fd) = F(fs) * F(ft); break; // mul.s
+		case 2: // mul.s
+			if ((my_isinf(F(fs)) && F(ft) == 0.0f) || (F(fs) == 0.0f && my_isinf(F(ft)))) {
```
Comment (hrydgard) [inline Core/MIPS/MIPSInt.cpp:978]: It's probably better to always have the my_isinf check on the left side of the & clauses - inf is a lot rarer than 0.0 so more short-circuiting.

(by the way, have you checked the NaN sign of negative inf * positive zero and positive inf * negative zero?

**Review by hrydgard — APPROVED** [review]
I guess if we get reports of perf regressions we'll have to gate it by game ID, but hopefully will not be a big deal in practice...

File: Core/MIPS/MIPSInt.cpp (hunk 1 above)
Comment (unknownbrackets) [reply Core/MIPS/MIPSInt.cpp]: Fair, changed.

Yes, the sign and order of operands make no difference, neither does the flush to zero flag.

-[Unknown]

**Pull request merged**